chore(terraform): update network module version to 10.0.0 to incorporate latest features and improvements

sachincool opened the pull request.

sachincool committed:
chore(terraform): update network module version to 10.0.0 to incorporate latest features and improvements

github-actions[bot] committed:
terraform-docs: automated action

sachincool committed:
chore(versions.tf): update google provider version to 6.21 to ensure compatibility with latest features and fixes

github-actions[bot] committed:
terraform-docs: automated action

dunefro commented on gcn.tf:
```diff
@@ -3,7 +3,7 @@
   count                   = var.use_existing_network ? 0 : 1
   source                  = "terraform-google-modules/network/google"
-  version                 = "9.3.0"
+  version                 = "10.0.0"
```
Lets take a careful look at this. Its a major version upgrade - we might have to test this thoroughly

sachincool replied:
I did the test, it was backward compatible. Terraform showed no diff.

sachincool committed:
feat(network): add network_profile variable to support custom network profiles

Add a new variable `network_profile` to allow specifying a network
profile URL during network creation. This enhances flexibility by
enabling the use of predefined network configurations.

github-actions[bot] committed:
terraform-docs: automated action
